internal/connector/internal/handlers/connector_cluster.go

```diff
@@ -244,22 +245,22 @@ func (o *ConnectorClusterHandler) buildAddonParams(serviceAccount *api.ServiceAc
 		},
 		{
 			Id:    "client-id",
-			Value: serviceAccount.ClientID,
+			Value: cluster.ClientId,
 		},
 		{
 			Id:    "client-secret",
-			Value: serviceAccount.ClientSecret,
+			Value: cluster.ClientSecret,
 		},
 	}
 	return p
 }
 
-func (o *ConnectorClusterHandler) buildTokenURL(serviceAccount *api.ServiceAccount) (string, error) {
+func (o *ConnectorClusterHandler) buildTokenURL(cluster dbapi.ConnectorCluster) (string, error) {
 	u, err := url.Parse(o.KeycloakConfig.KafkaRealm.TokenEndpointURI)
```

Comment:
"KeycloakConfig.KafkaRealm.TokenEndpointURI" this needs to be set on basis of which sso provider is used. `https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/pkg/client/keycloak/config.go#L89`

if the provider is MAS_SSO use the KafkaRealm.
If the provider is REDHAT_SSO use the RedhatSSORealm

https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/pkg/services/sso/keycloak.go#L26 will return the set sso service/provider realm